backend/calc.py

```diff
@@ -1,12 +1,23 @@
-# shitty proof of concept. run the run_test() method to try it (and maybe change some values).
-# todo: break this out into files properly
-
+"""
+calc.py
+Contains mathematical breakdowns of the stats used to calculate damage, and also contains a function
+(CharacterStats: calc_damage) to estimate the DPS of a character given the party composition and gear stats.
+"""
+from __future__ import annotations
 from enum import Enum, auto
 import itertools
 import math
+from typing import ClassVar
 
 
 class Stats(Enum):
+    """
+    Contains math factors for each individual stat.
+
+    base: the base value for each stat
+    m_factor: ???
```

Comment:
honestly this is fucking cursed. a lot of the damage formulas follow the formula:

`m_factor * (stat_value - base) + m_scalar`

this entire thing is basically a stat table to hide the magic math a bit

Reply:
noted and added